streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -175,13 +176,15 @@ public class KafkaStreams implements AutoCloseable {
     private final KafkaClientSupplier clientSupplier;
     protected final TopologyMetadata topologyMetadata;
     private final QueryableStoreProvider queryableStoreProvider;
+    private final StandbyUpdateListener delegatingStandbyUpdateListener;
 
     GlobalStreamThread globalStreamThread;
     private KafkaStreams.StateListener stateListener;
     private StateRestoreListener globalStateRestoreListener;
     private boolean oldHandler;
     private BiConsumer<Throwable, Boolean> streamsUncaughtExceptionHandler;
     private final Object changeThreadCount = new Object();
+    private StandbyUpdateListener globalStandbyListener;
```

Comment:
I know you're just following the established pattern here, but it always felt weird to me that we have two different top-level fields for the restore listener. I think it makes more sense for the "global" standby listener to just be completely encapsulated by the DelegatingStandbyUpdateListener class, and then make that class static. 

Imo we should also name it `userStandbyListener` rather than `globalStandbyListener` to avoid confusion with global state stores. I decided to just whip up a quick PR with all these fixes for the existing restore listener, so you can just take a look at https://github.com/apache/kafka/pull/14886 to see exactly what I'm suggesting we do for the standby listener

Reply:
I agree. Your proposal makes things easier to understand. Thanks! :)